routes/feed/downvote_post.js

```diff
@@ -1,62 +1,66 @@
-import { Router } from 'express'
-import Post from '../../models/Feed/Post.js'
-import Feed from '../../models/Feed/Feed.js'
+import { Router } from 'express';
+// import Post from '../../models/Feed/Post.js'
+// import Feed from '../../models/Feed/Feed.js'
 import { FEED_ID } from '../../index.js';
+import client from "../../models/prisma/prisma.js"
 
 const router = Router();
 
 router.post('/', async (req, res) => {
+  if (req.user.email.split('@')[0] !== req.body.onid) {
+    return res.status(401).json({ error: 'Unauthorized' });
+  }
 
-    if (req.user.email.split('@')[0] !== req.body.onid) {
-        return res.status(401).json({ error: 'Unauthorized' });
+  const { postId, isUpvoted, isDownvoted, onid } = req.body;
+
+  try {
+    const postToDownvote = await client.post.findUnique({
+      where: {
+        id: parseInt(postId),
+      },
+      includes: {
+        postedBy: true,
+      },
+    });
+    // console.log("postToDownvote before:", postToDownvote)
+
+    if (!postToDownvote) {
+      return res.status(400).json({ error: 'Post to downvote not found' });
     }
 
-    const { postId, isUpvoted, isDownvoted, onid } = req.body
+    let upvotes = [...new Set(postToDownvote.upvotes)];
+    let downvotes = [...new Set(postToDownvote.downvotes)];
 
-    try {
-        const postToDownvote = await Post.findById(postId)
-        // console.log("postToDownvote before:", postToDownvote)
+    const alreadyDownvoted = downvotes.includes(parseInt(onid));
 
-        if (!postToDownvote) {
-            return res.status(400).json({ error: 'Post to downvote not found' })
+    if (isDownvoted === false && !alreadyDownvoted) {
+      if (isUpvoted === true) {
+        // postToDownvote.upvotes -= 1
```

Comment:
not sure if this is here as a description or if it just dead code. If it is dead code that we can probably delete it